lib/Scheduler/hist_table.cpp

```diff
@@ -467,33 +493,106 @@ static bool doesHistoryPeakCostDominate(InstCount OtherPrefixCost,
   return LCE->GetBestCost() <= OtherPrefixCost;
 }
 
+// First pass simply compares current to history RP
+static bool doesHistoryPeakCostDominateFrstPss(InstCount OtherPrefixSpillCost,
+                                               InstCount HistPrefixSpillCost,
+                                               InstCount HistSuffixRPCost,
+                                               LengthCostEnumerator *LCE) {
+  if (OtherPrefixSpillCost >= HistPrefixSpillCost ||
+      HistSuffixRPCost > HistPrefixSpillCost)
+    return true;
+
+  return LCE->getBestSpillCost() <= OtherPrefixSpillCost;
+}
+
+static bool doesHistoryPeakCostDominateScndPss(InstCount OtherPrefixSpillCost,
+                                               InstCount HistPrefixSpillCost,
+                                               InstCount HistSuffixRPCost,
+                                               LengthCostEnumerator *LCE) {
+
+  // After propogating RP values for probed nodes that fail RP checks, this
+  // check will likely due more pruning
+  if (HistSuffixRPCost > LCE->getTrgtSpillCostConstrnt()) {
+    if (HistSuffixRPCost > HistPrefixSpillCost)
+      return true;
+
+    // if the above condition fails (e.g. HistPrefixSpillCost >=
+    // HistSuffixRPCost) then we will arrive at this assert. We also know that
+    // HistSuffixRPCost >= TrgtSpillCost due to outer condition. Thus, this
+    // would imply that we are currently in a condition where
+    // HistPrefixSpillCost > TrgtSpillCost. Based on the way we store history
+    // schedules, it is impossible to reach this condition. This assert should
+    // never be reached
+    assert(false);
```

Comment:
llvm::report_fatal_error? At a minimum, this should be `assert(false && "Some message making it possible to figure out which assert fired and giving a bit of information)`

Reply:
I figured someone would say something about this one -- I'll go with the fatal_error -- thanks